python/cudf/cudf/core/series.py

```diff
@@ -1815,6 +1815,27 @@ def drop_duplicates(self, keep="first", inplace=False, ignore_index=False):
     def fill(self, fill_value, begin=0, end=-1, inplace=False):
         return self._fill([fill_value], begin, end, inplace)
 
+    def fillna(self, value, method=None, axis=None, inplace=False, limit=None):
+        if isinstance(value, pd.Series):
+            value = Series.from_pandas(value)
+
+        if not (is_scalar(value) or isinstance(value, (dict, Series))):
```

Comment:
Let's use either Pandas' `is_dict_like` or `isinstance(value, abc.Mapping` to allow for other dict-like objects.